python/ray/data/context.py

```diff
@@ -82,7 +82,7 @@
 OK_PREFIX = "✔️ "
 
 # Default batch size for batch transformations.
-DEFAULT_BATCH_SIZE = 4096
+DEFAULT_BATCH_SIZE = "4096"
```

Comment:
Since the context-level default being a string that's eventually int-casted is a bit hacky, could we keep this as the `4096` int and have a `"default"` sentinel as the default arg in the `.map_batches()` signature, where we then internally do the following conversion?

```python
def map_batches(
    ...,
    batch_size = "default",
    ...,
):
    ...

    if batch_size == "default":
        batch_size = DEFAULT_BATCH_SIZE
    elif batch_size is not None:
        if batch_size < 1:
            raise ValueError("Batch size cannot be negative or 0")
        # Enable blocks bundling when batch_size is specified by caller.
        target_block_size = batch_size
```